modules/core/shared/src/main/scala/gem/EphemerisKey.scala

```diff
@@ -49,6 +49,10 @@ object EphemerisKey {
     */
   sealed abstract class Horizons(val queryString: String) extends EphemerisKey
 
+  /** Horizons designation for asteroids, old and new style.
+    */
+  sealed abstract class Asteroid(s: String) extends Horizons(s)
```

Comment:
In this particular case why `Asteroid` is not a `trait`? It's still not clear to me when to choose a trait or an abstract class.

Reply:
It can't be a trait because it extends an abstract class.  The `Horizons` class it extends is an abstract class because it contains a `queryString` field.  That could have been specified as a method and then implemented in the various instances but this seemed easier.